File: src/OpenCvSharp.Test/Samples/Snake.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace OpenCvSharp.Test
{
    /// <summary>
    /// 輪郭検出
    /// </summary>
    /// <remarks>http://opencv.jp/sample/object_tracking.html#snake</remarks>
    class Snake
    {        
        public Snake()
        {
            using (IplImage src = new IplImage(Const.ImageCake, LoadMode.GrayScale))
            using (IplImage dst = new IplImage(src.Size, BitDepth.U8, 3))
            {
                CvPoint[] contour = new CvPoint[100];
                CvPoint center = new CvPoint(src.Width / 2, src.Height / 2);
                for (int i = 0; i < contour.Length; i++)
                {
                    contour[i].X = (int)(center.X * Math.Cos(2 * Math.PI * i / contour.Length) + center.X);
                    contour[i].Y = (int)(center.Y * Math.Sin(2 * Math.PI * i / contour.Length) + center.Y);
                }
                Console.WriteLine("Press any key to snake\nEsc - quit");
                using (CvWindow w = new CvWindow())
                {
                    while (true)
                    {
                        src.SnakeImage(contour, 0.45f, 0.35f, 0.2f, new CvSize(15, 15), new CvTermCriteria(1), true);
                        src.CvtColor(dst, ColorConversion.GrayToRgb);
                        for (int i = 0; i < contour.Length - 1; i++)
                        {
                            dst.Line(contour[i], contour[i + 1], new CvColor(255, 0, 0), 2);
                        }
                        dst.Line(contour[contour.Length - 1], contour[0], new CvColor(255, 0, 0), 2);
                        w.Image = dst;
                        int key = CvWindow.WaitKey();
                        if (key == 27)
                        {
                            break;
                        }
                    }
                }
            }
        }
    }
}

File: sample/CStyleSamplesVB/Samples/Snake.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports OpenCvSharp

' Namespace OpenCvSharpSamplesVB
Imports SampleBase

''' <summary>
''' 輪郭検出
''' </summary>
''' <remarks>http://opencv.jp/sample/object_tracking.html#snake</remarks>
Friend Module Snake
    Public Sub Start()
        Using src As New IplImage(FilePath.Image.Cake, LoadMode.GrayScale), _
             dst As New IplImage(src.Size, BitDepth.U8, 3)
            Dim contour(99) As CvPoint
            Dim center As New CvPoint(src.Width \ 2, src.Height \ 2)
            For i As Integer = 0 To contour.Length - 1
                contour(i).X = CInt(Math.Truncate(center.X * Math.Cos(2 * Math.PI * i \ contour.Length) + center.X))
                contour(i).Y = CInt(Math.Truncate(center.Y * Math.Sin(2 * Math.PI * i \ contour.Length) + center.Y))
            Next i
            Form1.TextBox1.AppendText("Press any key to snake")
            Form1.TextBox1.AppendText(Environment.NewLine & "Esc - quit")
            Using w As New CvWindow()
                Do
                    src.SnakeImage(contour, 0.45F, 0.35F, 0.2F, New CvSize(15, 15), New CvTermCriteria(1), True)
                    src.CvtColor(dst, ColorConversion.GrayToRgb)
                    For i As Integer = 0 To contour.Length - 2
                        dst.Line(contour(i), contour(i + 1), New CvColor(255, 0, 0), 2)
                    Next i
                    dst.Line(contour(contour.Length - 1), contour(0), New CvColor(255, 0, 0), 2)
                    w.Image = dst
                    Dim key As Integer = CvWindow.WaitKey()
                    If key = 27 Then
                        Exit Do
                    End If
                Loop
            End Using
        End Using
    End Sub
End Module
' End Namespace
